### DISCUSSION TITLE
PDF Consolidation & Numbering Scheme

### DISCUSSION THREAD

**Discussion by christopherball** [spamegg1/Epp-Discrete-Math-5th-solutions#7] [General]
Would it be possible to match the exact numbering scheme of the book?  For me, I find it a bit jarring that every problem for every set of each PDF starts over from 1.x.x.

The book conforms to the following:
`[Chapter]`.`[Set]`.`[Problem]`.`[Part]`

So if we randomly take a look at the `chapter3.pdf` for example, rather than seeing:
![image](https://github.com/spamegg1/Epp-Discrete-Math-5th-solutions/assets/5790394/df1b7c4c-1318-4060-8018-0f3e4b06aead)

We would instead simply see:
```
Chapter 3 - The Logic of Quantified Statements

3.1.1 
A menagerie consists of seven brown dogs, two black dogs, six gray cats, ten black cats, five blue birds, six yellow birds, and one black bird. Determine which of the following statements are true and which are false.

3.1.1.a 
There is an animal in the menagerie that is red.

Proof. False (brown, black, gray, blue, yellow, but no red animals).

3.1.1.b
...
```

Doing so would make it easier to simply combine all the PDF's together and more easily figure out which chapter/set/problem you're looking at without needing to keep the table of contents around or scrolling up to check which set you're in.

**Comment by spamegg1** [comment]
Great idea! I'll look into it. Probably something I have to change in my LaTeX code / settings to achieve it. If there is no way to do that, I can add all the code into a single source file and compile that.

**Reply by spamegg1** [reply]
I am close to finishing Chapter 12 and the whole book, after that I'll try putting everything in one file.

**Reply by christopherball** [reply]
Out of curiosity, what do you think of the book now that you're reaching the end?  I just recently picked up a hardcover copy of it with the plan to review the material and see if there's areas I've forgotten (or perhaps never learned properly).  Between this book and Book of Proof, I can't imagine better material on the subject.

**Reply by spamegg1** [reply]
Agreed, it's great, I'd like to base my own math + CS curriculum on it. Although the exercises and answers have quite a lot of typos. http://condor.depaul.edu/~sepp/DM5eErrata.pdf https://academic.cengage.com/resource_uploads/downloads/1337694193_658745.pdf

One strange part is that finite automata is at the very end, in the final Chapter. I'd like to pair it with Structural Induction (end of Chapter 5) instead. This is also how MIT Math for CS does it because state machines + structural induction go together.

**Reply by christopherball** [reply]
Yeah I grabbed a copy of the latest errata before purchasing the physical book - figured I'd simply handwrite in the corrections as I went along.  Frankly any book this size will have typos along the way.

With respect to chapter 12, the `preface xx` page shows a chapter flow tree where it mentions that both chapter 10 and 12 (mostly) could be a learning path directly following 5.

**Comment by AhmedAzor** [comment]
Out of curiosity too, why do you learn proofs?
check this out https://scholarworks.gvsu.edu/books/24/ , I didn't read it,
maybe after Epp but seems good.

**Reply by spamegg1** [reply]
Not necessary, Epp covers all of that book (in more detail).

**Comment by spamegg1** [comment]
OK, I'm finished with the whole book now. It turns out that this numbering scheme is not possible, there is no way to add chapter numbers at the beginning of the numbers, at least not without incredibly massive rewriting of the source code. I would have to write some clever script to rewrite it. 

There are other issues as well; the table of contents becomes massive (over a hundred pages), the PDF size becomes massive, compilation takes ridiculously long, and some issues with the LaTeX packages / fonts from different files conflicting with each other.

So we'll have to live with the separate chapter PDFs for now. Sorry about that!

**Comment by spamegg1** [comment]
Alright, I was able to figure out the source editing, and put everything in one PDF with chapter / section / exercise numbering. 

Check out https://github.com/spamegg1/Epp-Discrete-Math-5th-solutions/blob/master/all.pdf @christopherball Does it have the numbering you wanted?

I had to remove the Table of Contents. The PDF still has an index that PDF readers can understand, so that should be good enough! 

I also did not check if all the images, tables, colors, fonts etc. work correctly. The compilation is successful without any errors or warnings but some things might be out of place. Some images might show up on the next page instead of where they should be.

It's 18 MBs and 1088 pages (without ToC) :scream: Line count is 56512! :muscle:

**Reply by christopherball** [reply]
Awesome updates!  While I am out of pocket with my phone for a bit, I spot checked areas across the first few hundred pages and didn't notice any missing graphics or the likes.  The consolidated pdf certainly functions well as a solutions comparison resource.  Will periodically compare my results to your own to look for any notable discrepancies.  Cheers!
